dataactcore/models/baseInterface.py

```diff
@@ -157,3 +159,10 @@ def getNameFromDict(self, model, dictName, fieldName, fieldValue, idField):
                 return key
         # If not found, raise an exception
         raise ValueError("Value: " + str(fieldValue) + " not found in dict: " + str(dict))
+
+
+@contextmanager
+def databaseSession():
+    interface = BaseInterface()
+    yield interface.session
+    interface.close()
```

Comment:
Gotcha, so we'd see something like:
```python
interface = BaseInterface()
interface.connection    # not null
with databaseSession() as sess:
    doSomeThingWith(sess)
interface.connection    # now null
```

What do you think about skipping that shared connection altogether? Say we separate out the engine/connection/session creation into a separate function:
```python
class BaseInterface:
    def __init__(self):
        self.engine, self.connection, self.session = newDatabaseContext()
        ...
```
Then we can choose not to use those static variables:
```python
def databaseSession():
    engine, connection, session = newDatabaseContext()
    yield session
    engine.dispose()
    connection.close()
    session.close()
```
This'd recreate and close the database connection with each `databaseSession` call, but we'd explicitly know when the connections were valid (i.e. through the end of the context). This might be easier to reason about than the static vars + garbage collection approach.